sfs/mono/source.py

```diff
@@ -705,6 +705,80 @@ def plane_averaged_intensity(omega, x0, n0, grid, c=None):
     return util.XyzComponents([i * n for n in n0])
 
 
+def pulsating_sphere(omega, x0, radius, amplitude, grid, c=None):
+    """Sound pressure of a pulsating sphere.
+
+    Parameters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    x0 : (3,) array_like
+        Center of sphere.
+    radius : float
+        Radius of sphere.
+    amplitude : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    numpy.ndarray
+        Sound pressure at positions given by *grid*.
+    """
+    if c is None:
+        c = defs.c
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    grid = util.as_xyz_components(grid)
+
+    r = np.linalg.norm(grid - x0)
+    r[r <= radius] = np.nan
+    theta = np.arctan(1, k * r)
+    impedence = defs.rho0 * c * np.cos(theta) * np.exp(1j * theta)
+    return impedence * 1j * omega * amplitude * radius / r \
+        * np.exp(-1j * k * (r - radius))
+
+
+def pulsating_sphere_velocity(omega, x0, radius, amplitude, grid, c=None):
+    """Particle velocity of a pulsating sphere.
+
+    Parameters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    x0 : (3,) array_like
+        Center of sphere.
+    radius : float
+        Radius of sphere.
+    amplitude : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    `XyzComponents`
+        Particle velocity at positions given by *grid*.
+    """
+    if c is None:
+        c = defs.c
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    offset = grid - x0
+    r = np.linalg.norm(offset)
+    r[r <= radius] = np.nan
+    return util.XyzComponents([1j * omega * amplitude * radius / r**2
+                               * np.exp(-1j * k * (r - radius)) * o
+                               for o in offset])
```

Comment:
/home/mg/git/sfs-python/sfs/mono/source.py:779: RuntimeWarning: invalid value encountered in true_divide
      for o in offset])

See above.